lib/pf/util.pm

```diff
@@ -1011,6 +1012,28 @@ sub cert_has_expired {
     return time > $expiration;
 }
 
+=item strip_username
+
+Will strip a username matching pattern user@realm or \\realm\user
+
+Returns ($user,$realm) if found or ($user) if not matching any realm pattern
+
+=cut
+
+sub strip_username {
+    my ($username) = @_;
+    if($username =~ /(.*)\@(.*)/){
+        return ($1,$2);
+    }
+    elsif($username =~ /(.*)\%(.*)/){
+        return ($1,$2);
+    }
+    elsif($username =~ /\\\\(.*)\\(.*)/) {
+        return ($2,$1);
+    }
+    return $username;
+}
+
```

Comment:
I would agree with you if the regex was used more than once :) Since it is a pretty simple sub, almost self-described, I think just highlighting the use cases with comment would be enough.
Anyway.. one way or another... (https://www.youtube.com/watch?v=1VFuHj9_Tgw)
I think that comment would be necessary even in the case of a named regex.